javascript/javascript2/week2/index.js

```diff
@@ -0,0 +1,137 @@
+// Doubling of number
+// let numbers = [1, 2, 3, 4];
+// let newNumbers = [];
+
+// for (let i = 0; i < numbers.length; i++) {
+//     if (numbers[i] % 2 !== 0) {
+//         newNumbers[i] = numbers[i] * 2;
+//     }
+// }
+
+// console.log("The doubled numbers are", newNumbers); // [2, 6]
+
+const numbers = [1, 2, 3, 4];
+
+const doubledOddNumbers = numbers
+    .filter((item) => item % 2 !== 0)
+    .map((item) => item * 2);
+console.log("The doubled numbers are " + doubledOddNumbers);
+
+// Working with movies
+
+//1. Movies with a short title
+
+const moviesWithAShortTitle = movies.filter((item) => item.title.length < 3);
+console.log(moviesWithAShortTitle);
+
+// 2. An array of movie titles with long movie titles
+
+const moviesWithALongTitles = movies
+    .filter((item) => item.title.length > 60)
+    .map((item) => item.title);
+
+console.log(moviesWithALongTitles);
+
+// 3. Count the number of movies made between 1980-1989 (including both the years)
+
+const moviesFromEighties = movies.filter(
+    (movie) => movie.year <= 1989 && movie.year >= 1980
+).length;
+
+console.log(moviesFromEighties);
+
+// 4. Create a new array that has an extra key called tag.
+//  The tag is based on the rating: Good (>= 7), Average (>= 4 and < 7), Bad (< 4)
+
+const newTag = movies.map((item) => {
+    if (item.rating >= 7) {
+        item.tag = "Good";
+        return item;
+    } else if (item.rating >= 4 && item.rating < 7) {
```

Comment:
Since the conditions in if statement are being evaluated from top to bottom, the second condition will get evaluated only if the the rating doest not fit the first one ie. is smaller than 7 and therefore you can save some key strokes by not writing the multiple condition in your "else if"💃